lib/rubocop/cop/rails/where_not_with_multiple_arguments.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop identifies calls to where.not with multiple hash arguments.
+      #
+      # The behavior of where.not changed in Rails 6.1. Prior to the change,
+      # `.where.not(trashed: true, role: 'admin')` evaluated to
+      # `WHERE trashed != TRUE AND role != 'admin'`.
+      # From Rails 6.1 onwards, this executes the query
+      # `WHERE NOT (trashed == TRUE AND roles == 'admin')`.
+      #
+      # @example
+      #   # bad
+      #   User.where.not(trashed: true, role: 'admin')
+      #   User.where.not(trashed: true, role: ['moderator', 'admin'])
+      #   User.joins(:posts).where.not({ posts: { trashed: true, title: 'Rails' } })
+      #
+      #   # good
+      #   User.where.not(trashed: true)
+      #   User.where.not(role: ['moderator', 'admin'])
+      #   User.where.not(trashed: true).where.not(role: ['moderator', 'admin'])
+      #   User.where.not('trashed = ? OR role = ?', true, 'admin')
+      class WhereNotWithMultipleArguments < Base
```

Comment:
Maybe it's a more specific better name.

```suggestion
      class WhereNotWithMultipleConditions < Base
```